src/vstest.console/Internal/ConsoleLogger.cs

```diff
@@ -147,6 +147,12 @@ public void Initialize(TestLoggerEvents events, string testRunDirectory)
             events.TestRunMessage += this.TestMessageHandler;
             events.TestResult += this.TestResultHandler;
             events.TestRunComplete += this.TestRunCompleteHandler;
+
+            // Register for the discovery events.
```

Comment:
Fix for https://github.com/Microsoft/vstest/issues/1569. ported from https://github.com/Microsoft/vstest/pull/1111/

/cc @abhishkk